src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
+    dateSet.YYYY = '20' + `${dateSet.YY}`;
+  } else {
+    dateSet.YYYY = '19' + `${dateSet.YY}`;
+  }
+
+  for (let i = 0; i < 3; i++) {
+    newDate.push(dateSet[toFormat[i]]);
+  }
+
+  return newDate.join(toFormat[toFormat.length - 1]);
```

Comment:
There's a potential issue with the join operation. The 'toFormat' array's last element is assumed to be the separator, but there's no check to ensure that 'toFormat' has a separator as its last element. This could lead to errors if 'toFormat' does not follow the expected structure. A validation or a more robust way of determining the separator could prevent such issues.